File: Python/Problems/Problems/problem3746.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter

from .problem import Problem

class Problem3746(Problem):
    def test(self) -> bool:
        s = "aabbab"

        expected = 0

        result = self.minLengthAfterRemovals(s)

        print(result)

        return result == expected

    def minLengthAfterRemovals(self, s: str) -> int:
        na = 0
        for c in s:
            if c == 'a':
                na += 1

        return abs(len(s) - 2 * na)

File: C++/Problems/Problems/problem_3746.cpp
#include "problem_3746.h"

bool problem_3746::test() {
    std::string s = "aabbab";

    int expected = 0;

    auto result = minLengthAfterRemovals(s);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_3746::minLengthAfterRemovals(std::string s) {
    int na = 0;
    for (auto c : s)
    {
        if (c == 'a')
            na++;
    }

    return std::abs((int)s.size() - 2 * na);
}
